http-server/src/server.rs

```diff
@@ -302,116 +302,25 @@ impl<M: Middleware> Server<M> {
 					// two cases: a single RPC request or a batch of RPC requests.
 					async move {
 						if let Err(e) = access_control_is_valid(&access_control, &request) {
```

Comment:
this became hard to review, however validate origins and headers before checking the method kind.